Fix relative path in urdf

fabinsch opened the pull request:
Currently, we cannot import a urdf that specifies a relative mesh path without using `//package`, see https://github.com/stack-of-tasks/pinocchio/issues/1741.

jcarpent commented on src/parsers/utils.hpp:
```diff
@@ -99,6 +99,18 @@ namespace pinocchio
         throw std::invalid_argument(exception_message);
       }
     }
+    else if (string.substr(0,2).compare("./") == 0)
```
This seems not to be robust depending on the OS. Could you use rather Boost.Filesystem for that?
Additionally, it will be nice to add a test of the fix.

fabinsch replied:
thanks for the hint. I adapted it to use `Boost.Filesystem`.

jcarpent reviewed (commented):
I think we should also link the main pinocchio lib to boost.filesystem, which seems only the case for Windows. Could you confirm?

fabinsch commented:
> I think we should also link the main pinocchio lib to boost.filesystem, which seems only the case for Windows. Could you confirm?

It's fine, we are already linking to boost.filesystem for all platforms (https://github.com/stack-of-tasks/pinocchio/blob/master/src/CMakeLists.txt#L60).

fabinsch commented:
@wxmerkt we have a problem with the [ROS CI](https://github.com/stack-of-tasks/pinocchio/runs/8225315773?check_suite_focus=true#step:4:544) and to me it seems that the eigenpy version are not the good ones. Could you have a look please or give me a hint what I can do to fix it ? Thank you

wxmerkt commented:
> @wxmerkt we have a problem with the [ROS CI](https://github.com/stack-of-tasks/pinocchio/runs/8225315773?check_suite_focus=true#step:4:544) and to me it seems that the eigenpy version are not the good ones. Could you have a look please or give me a hint what I can do to fix it ? Thank you

The EigenPy versions on ROS Melodic/18.04 were reverted to an older release since dynamic-graph hadn't been fixed / re-released. Once dynamic-graph/SoT/TSID are ready for release, we can re-release EigenPy & HPP-FCL on Melodic.

@nim65s have you created releases to Melodic with the fixed packages? If so, I can get EigenPy & HPP-FCL back to the latest version

nim65s commented:
The releases are done, but now they require recent versions of eigenpy… Can you already put back the latest eigenpy versions ? Or how should we proceed ?

fabinsch commented:
I've added a unittest @jcarpent . I looked into using the `pinocchio::urdf::buildModelFromXML` function in order to not add a "new" urdf file but for now we are not loading geometry objects with this function. Therefore I added another urdf which is a copy of the simple humanoid and I changed to path to the mesh from `"package://simple_humanoid_description/box.stl"` to `"./simple_humanoid_description/box.stl"`. Test is passing locally.

wxmerkt commented:
EigenPy/HPP-FCL release tracker for 18.04/Melodic: https://github.com/ros/rosdistro/pull/34371
